applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_3d.cpp

```diff
@@ -171,6 +171,136 @@ void ElasticIsotropic3D::FinalizeMaterialResponseCauchy(ConstitutiveLaw::Paramet
 /***********************************************************************************/
 /***********************************************************************************/
 
+bool ElasticIsotropic3D::Has(const Variable<bool>& rThisVariable)
+{
+    if (rThisVariable == REQUIRES_MATERIAL_INITIALIZATION) {
+        return true;
+    } else if (rThisVariable == REQUIRES_MATERIAL_FINALIZATION) {
+        return true;
```

Comment:
I don't fully understand where this is going, but since you say it is necessary I could imagine that we could make this a flag of the CLaw-parameters
@loumalouomega / @RiccardoRossi what do you think abt making it a flag?

Reply:
honestly i don't love this.

If for performance reason we decide to go this way than we should assume that a CL by default requires both initialization and finalization. (i agree that it would be best to define them as CL's local flags).
This would guarantee that any unmodified law works exactly as before.

This could be achieved for example by implementing the get value in the base constitutive law class.

a change of this type should be anyway discussed both in the @KratosMultiphysics/design-committee  and @KratosMultiphysics/technical-committee